src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2138,28 +2178,42 @@ LogicalResult ONNXConvTransposeOp::inferShapes(
   auto padsOpt = pads();
   auto outputPads = output_padding();
   auto outputShape = output_shape();
-  // TODO: handle the spatial dimension computation if output shape is
-  // specified
-  assert(!outputShape.has_value() && "unhandled option in ConvTranspose");
-
-  // First two output dimensions consist of the number of batches and the
-  // number of kernels being applied.
-  SmallVector<int64_t, 4> outputDims;
-  // Insert batch size.
-  outputDims.emplace_back(xShape[0]);
-  // Insert number of filters being applied (number of output channels *
-  // groups).
-  outputDims.emplace_back(outChannels);
-  // Compute and insert spatial dims.
-  insertConvTransposeSpatialDim(outputDims, xShape, kernelShape, padsOpt,
-      stridesOpt, outputPads, outputShape, dilationsOpt);
+  llvm::SmallVector<int64_t, 4> outputShapeFinal;
 
-  // Set the output shape if it's not already set
-  if (!outputShape.has_value()) {
-    output_shapeAttr(builder.getI64ArrayAttr(outputDims));
+  if (outputShape.has_value()) {
+    if (xShape[0] != ArrayAttrIntVal(outputShape, 0)) {
+      return emitError("mismatch in batch size");
```

Comment:
nit: emitOpError is better because it helpfully prints the op name

same in line 2188 below

Reply:
I didn't know that `emitOpError` was better than `emitError`. Thanks for the info